api/v1alpha1/contour_types.go

```diff
@@ -188,6 +188,28 @@ type EnvoyNetworkPublishing struct {
 	// +kubebuilder:default={scope: External, providerParameters: {type: AWS}}
 	LoadBalancer LoadBalancerStrategy `json:"loadBalancer,omitempty"`
 
+	// NodePorts is a list of network ports to expose on each node's IP at a static
+	// port number using a NodePort Service. Present only if type is NodePortService.
+	// A ClusterIP Service, which the NodePort Service routes to, is automatically
+	// created. You'll be able to contact the NodePort Service, from outside the
+	// cluster, by requesting <NodeIP>:<NodePort>.
+	//
+	// For additional information on the NodePort Service, see:
+	//
+	//  https://kubernetes.io/docs/concepts/services-networking/service/#nodeport
+	//
+	// Names and port numbers must be unique in the list. Two ports must be specified,
+	// one named "http" for Envoy's insecure service and one named "https" for Envoy's
+	// secure service.
```

Comment:
It's not clear from this if the ports must always be specified, given that `omitempty` and `+optional` are also specified. This could use some clarification.

Reply:
PR updated to resolve ^